main/lsp/requests/hover.cc

```diff
@@ -59,6 +59,18 @@ LSPResult LSPLoop::handleTextDocumentHover(unique_ptr<core::GlobalState> gs, con
         } else if (auto defResp = resp->isDefinition()) {
             response->result = make_unique<Hover>(formatRubyCode(
                 clientHoverMarkupKind, methodDetail(*gs, defResp->symbol, nullptr, defResp->retType.type, nullptr)));
+        } else if (auto constResp = resp->isConstant()) {
+            const auto &data = constResp->symbol.data(*gs);
+            auto type = constResp->retType.type;
+            if (data->isClass()) {
```

Comment:
I had to cargo-cult this logic from `environment.cc` to get hover to work correctly on class/module/type aliases.

Before, a reference to `Foo` might show up as `Foo` or `T.untyped` rather than `T.class_of(Foo)`. Similarly, a reference to `T.type_alias(Foo)` would show up as `Foo` rather than `<Type: Foo>`.

There's other logic in `environment.cc` concerning fields and static fields that I did not cargo cult.

Reply:
I don't think you need the second branch(`isTypeAlias` one)

Reply:
it also failed ENFORCE, so you'd want to delete it :-)